backend/Api/Common/StorageService.cs

```diff
@@ -0,0 +1,76 @@
+using Core.DomainModels;
+using Database.DatabaseContext;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Api.Common;
+
+public class StorageService
+{
+    private readonly IMemoryCache _cache;
+    private readonly ApplicationContext _dbContext;
+
+    public StorageService(IMemoryCache cache, ApplicationContext context)
+    {
+        _cache = cache;
+        _dbContext = context;
+    }
+
+    public List<Consultant> LoadConsultants(string orgUrlKey)
+    {
+        if (_cache.TryGetValue<List<Consultant>>("asdf", out var consultants))
+            if (consultants != null)
+                return consultants;
+
+        var loadedConsultants = LoadConsultantsFromDb(orgUrlKey);
+        _cache.Set("asdf", loadedConsultants);
+        return loadedConsultants;
+    }
+
+    private List<Consultant> LoadConsultantsFromDb(string orgUrlKey)
```

Comment:
This was very clean! 👏 👏 But should LoadConsultants method be defined elsewhere?

Reply:
Another note: Should the naming "LoadConsultants" be something else, since its containing a lot of other stuff too? I mean this is not just a generic list of consultants, but has a lot of extended data attached to it.

Reply:
It looks like much, but it's really only loading data. 
All the logic in the database really just comes from avoiding a mega-join, but instead appending staffing, plannedabsence and vacations later